rust_snuba/src/processors/replays.rs

```diff
@@ -69,90 +77,100 @@ pub fn process_message(
                 tags_value.push(tag.1);
             }
 
-            // Compute user value.
-            let user = if !event.user.user_id.is_empty() {
-                event.user.user_id.clone()
-            } else if !event.user.username.is_empty() {
-                event.user.username.clone()
-            } else if !event.user.email.is_empty() {
-                event.user.email.clone()
-            } else if let Some(ip) = event.user.ip_address {
-                ip.to_string()
-            } else {
-                String::new()
+            // Unwrap the ip-address string.
+            let ip_address_string = event.user.ip_address.unwrap_or_default();
+            let (ip_address_v4, ip_address_v6) = match ip_address_string.parse::<IpAddr>() {
+                Err(_) => (None, None),
+                Ok(ip) => match ip {
+                    IpAddr::V4(ipv4) => (Some(ipv4), None),
+                    IpAddr::V6(ipv6) => (None, Some(ipv6)),
+                },
```

Comment:
```suggestion
                Ok(IpAddr::V4(ipv4)) => (Some(ipv4), None),
                Ok(IpAddr::V6(ipv6)) => (None, Some(ipv6)),
```

You can also match sub-patterns